flipperdevices/flipperzero-firmware
# picopass read bug fixes

# What's new

- fix checking user elite keys - current behaviour checks only the default dictionary
- include calculated Kd when saving cards - current behaviour sets the Kd block to FFFFFFFFFFFFFFFF in the save file (this is an important pre-requisite for emulation which is coming soon)

# Verification 

- Attempt to read an elite keyed iClass card with a key not in the default (`picopass/assets/iclass_elite_dict.txt`) dictionary
- Observe the read fails
- Add the key to the user dictionary (`picopass/assets/iclass_elite_dict_user.txt`)
- Try to read the card again
- Observe the read succeeds
- Save the read card
- Observe the card-specific derived key in block 3 in the saved file rather than all FF's

# Checklist (For Reviewer)
